src/hooks/toast/useErrorToast.tsx

```diff
@@ -0,0 +1,41 @@
+import { PropsWithChildren } from "react"
+import toast from "react-hot-toast"
+import { v4 as uuid } from "uuid"
+
+import { ToastProps } from "@/hooks/toast/useLoadingToast"
+
+import { ErrorToast } from "@/components/Toast"
+
+type ErrorToastProps = PropsWithChildren<ToastProps>
+
+export function useErrorToast({
```

Comment:
There are a few problems with this design IMO. 

1. You've set it up in a way where we need to pass in the children when the hook is initialized when we will need to fire off toasts imperatively in callbacks. In this case, you'd want to have `fireToast` accept the message instead of the hook itself. 

2. Having separate hooks for every toast type is going to be tedious during implementation. It's going to be very common to need to handle multiple outcomes (success, error, waiting) in a single component and to do so you'll need to write out every time:

```tsx
const errorToast = useErrorToast()
const successToast = useSuccessToast()
const loadingToast = useLoadingToast()
```

I would propose a hook like this instead:

```tsx
// src/hooks/useToast.ts

// we don't need to set this on a per-toast basis; just set it once and use it for all of them
// we also don't need to set position. every toast will be in the same place; hardcode it
const duration = 7000

export const useToast = () => ({
  error: (message: string) =>
    toast.custom(
      (t) => (
        // note: t.id is provided, you don't need uuid()
        <ErrorToast isVisible={t.visible} onDismiss={() => toast.dismiss(t.id)}>
          {message}
        </ErrorToast>
      ),
      { duration }
    ),
  success: (message: string, txHash: string) => toast.custom((t) => <></>),
  loading: (message: string, txHash: string) => toast.custom((t) => <></>),
})
```

Then to implement: 

```tsx
// src/components/Vault/VaultDepositForm.tsx

// ...
const toast = useToast()

// ...
const waitDepositUnderlying = useWaitForTransaction({
  hash: depositUnderlying.data?.hash,
  onError: () => toast.error("Deposit failed"),
  onSuccess: (data) => toast.success("Deposit complete", data.transactionHash),
})
```

Reply:
I see, the only problem (if we can call it a problem) is the id management. Because the Loading/Success/Error should share the same id to allow the waterfall between them if you want, if don't they will be displayed as 3 distinct toasts.

Reply:
I also think the string as a parameter doesn't handle every use case. For example, if you want to display an arbiscan or etherscan link to your error message, it won't work. Additionally, passing a transaction hash as a parameter is specific to one use case of the toast utilization; what if you want to use it for another use case?

Reply:
The ID issue is not a problem because you should dismiss the loading toast before showing an error/success toast. 

On the children prop vs specific props (message, txHash), I would ask you: what other use case do you see us having for these?

It's better to build something specific for the use cases we have and expand upon it later than to try to make something that is more open ended but harder to use when implementing. If we need a new toast type we can simply add a new toast type. 

And yeah, the tx waiting thing is very specific -- but -- it is optimized for a common use case in our application. I ask again what else would we be showing a loading toast for if it isn't waiting for a tx to confirm or be submitted?

I would design a specific toast around that workflow: waiting for user to sign and then waiting for the tx to confirm.